sugar.go

```diff
@@ -0,0 +1,214 @@
+// ErrorLevel. It doesn't add to the message's structured context.
+func (s *SugaredLogger) Errorf(template string, args ...interface{}) {
+	if ce := s.core.Check(ErrorLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// DPanic logs a message and some key-value pairs using the underlying logger's
+// DPanic method. Keys and values are treated as they are in the With
+// method. (See Logger.DPanic for details.)
+func (s *SugaredLogger) DPanic(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(DPanicLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// DPanicf uses fmt.Sprintf to construct a dynamic message, which is passed to
+// the underlying Logger's DPanic method. (See Logger.DPanic for details.) It
+// doesn't add to the message's structured context.
+func (s *SugaredLogger) DPanicf(template string, args ...interface{}) {
+	if ce := s.core.Check(DPanicLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// Panic logs a message and some key-value pairs at PanicLevel, then panics.
+// Keys and values are treated as they are in the With method.
+func (s *SugaredLogger) Panic(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(PanicLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// Panicf uses fmt.Sprintf to construct a dynamic message and logs it at
+// PanicLevel, then panics. It doesn't add to the message's structured context.
+func (s *SugaredLogger) Panicf(template string, args ...interface{}) {
+	if ce := s.core.Check(PanicLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+// Fatal logs a message and some key-value pairs at FatalLevel, then calls
+// os.Exit(1). Keys and values are treated as they are in the With method.
+func (s *SugaredLogger) Fatal(msg interface{}, keysAndValues ...interface{}) {
+	if ce := s.core.Check(FatalLevel, sweetenMsg(msg)); ce != nil {
+		ce.Write(sweetenFields(keysAndValues, s.core)...)
+	}
+}
+
+// Fatalf uses fmt.Sprintf to construct a dynamic message and logs it at
+// FatalLevel, then calls os.Exit(1). It doesn't add to the message's
+// structured context.
+func (s *SugaredLogger) Fatalf(template string, args ...interface{}) {
+	if ce := s.core.Check(FatalLevel, fmt.Sprintf(template, args...)); ce != nil {
+		ce.Write()
+	}
+}
+
+func sweetenFields(args []interface{}, errLogger Logger) []zapcore.Field {
+	if len(args) == 0 {
+		return nil
+	}
+	if len(args)%2 == 1 {
+		errLogger.DPanic(oddNumberErrMsg, Any("ignored", args[len(args)-1]))
+	}
+
+	fields := make([]zapcore.Field, len(args)/2)
+	for i := range fields {
+		key := sweetenMsg(args[2*i])
+		fields[i] = Any(key, args[2*i+1])
+	}
+	return fields
+}
+
+func sweetenMsg(msg interface{}) string {
+	if str, ok := msg.(string); ok {
+		return str
+	}
+	return fmt.Sprint(msg)
+}
```

Comment:
In order:

* That's true. However, if you're using the sugared logger, I don't think a single allocation is breaking the bank.
* `fmt.Stringer` is the other obvious interface we should support, but yes, that's probably it. `logger.Error(err)` is so common that this seems worthwhile to me. This also makes debug-level logging much easier in development, since you can easily log arbitrary structs.
* Logging in most dynamically typed languages works this way, and that shapes many developers' expectations.